test.sh

```diff
@@ -10,9 +10,7 @@ test_err() {
 }
 
 test_js() {
-    if [[ ! -x ./node_modules/karma/bin/karma || ! -x ./node_modules/jshint/bin/jshint ]]; then
-        npm install
-    fi
+    karma --version && jshint --version || npm install
```

Comment:
npm is quite the pain. I think we may need to install globally always. There doesn't seem to be a way to ask npm if a package is installed either globally or locally. You have to query each separately, which is annoying.

If we assume a global install, then we might as well just add it to sirepo-dev. This would eliminate the need for a check here.

To answer Evan's specific question:

```bash
local p
if ! [[ $(type -p karma) && $(type -p jshint) ]]; then
     npm install
fi
```

You don't need a `>& /dev/null` this way. 

Do you both agree?